x-pack/plugins/reporting/server/export_types/printable_pdf/lib/get_custom_logo.ts

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { ReportingCore } from '../../../';
+import { UI_SETTINGS_CUSTOM_PDF_LOGO } from '../../../../common/constants';
+import { ConditionalHeaders } from '../../../types';
+
+export const getCustomLogo = async (
+  reporting: ReportingCore,
+  conditionalHeaders: ConditionalHeaders,
+  spaceId?: string
+) => {
+  const fakeRequest = reporting.getFakeRequest({ headers: conditionalHeaders.headers }, spaceId);
+  const uiSettingsClient = await reporting.getUiSettingsClient(fakeRequest);
+
+  const logo: string = await uiSettingsClient.get(UI_SETTINGS_CUSTOM_PDF_LOGO);
```

Comment:
```suggestion
  const [
    uiSettingsClient,
    logo,
  ] = await Promise.all([
    reporting.getUiSettingsClient(fakeRequest),
    uiSettingsClient.get(UI_SETTINGS_CUSTOM_PDF_LOGO),
  ];
```

Reply:
On second look, we can't do this since `uiSettingsClient` is needed by the 2nd statement and provided by the 1st. Sorry, I had to revert it.

Reply:
Shoot, I mistakingly just looked at the arguments, you're right I'll be more cautious next time :)